core/src/main/scala/org/apache/spark/BarrierCoordinator.scala

```diff
@@ -130,19 +139,37 @@ private[spark] class BarrierCoordinator(
 
     // Process the global sync request. The barrier() call succeed if collected enough requests
     // within a configured time, otherwise fail all the pending requests.
-    def handleRequest(requester: RpcCallContext, request: RequestToSync): Unit = synchronized {
-      val taskId = request.taskAttemptId
-      val epoch = request.barrierEpoch
+    private def handleRequest(
```

Comment:
why don't we still accept the `RequestToSync` here? IMO we should avoid passing many params around.

Reply:
+1